src/inputs/Table/ColumnTypes.js

```diff
@@ -0,0 +1,152 @@
+      newValue = DateUtils.strMax(newValue, minValue, dateFormat) || params.oldValue;
+    }
+    if (maxValue !== undefined) {
+      newValue = DateUtils.strMin(newValue, maxValue, dateFormat) || params.oldValue;
+    }
+  }
+  // Force date format before setting it
+  params.data[params.colDef.field] = DateUtils.format(DateUtils.parse(newValue, dateFormat), dateFormat);
+  return true;
+};
+
+const _intSetter = (params) => {
+  let newValue = parseInt(params.newValue);
+
+  if (!params.context.editMode || isNaN(newValue) || !Number.isSafeInteger(newValue)) {
+    newValue = params.oldValue;
+  } else {
+    // Min & max values are currently limited by the default cellEditor behavior
+    const DEFAULT_MIN_INT = -1e21 + 1;
+    const DEFAULT_MAX_INT = 1e21 - 1;
+    const configMinValue = params.column.userProvidedColDef?.cellEditorParams?.minValue;
+    const configMaxValue = params.column.userProvidedColDef?.cellEditorParams?.maxValue;
+    const minValue = configMinValue !== undefined ? configMinValue : DEFAULT_MIN_INT;
+    const maxValue = configMaxValue !== undefined ? configMaxValue : DEFAULT_MAX_INT;
+    newValue = Math.max(newValue, minValue);
+    newValue = Math.min(newValue, maxValue);
+    newValue = newValue.toString();
+  }
+  params.data[params.colDef.field] = newValue;
+  return true;
+};
+
+const _numberSetter = (params) => {
+  let newValue = parseFloat(params.newValue);
+  if (!params.context.editMode || isNaN(newValue)) {
+    newValue = params.oldValue;
+  } else {
+    const minValue = params.column.userProvidedColDef?.cellEditorParams?.minValue;
+    const maxValue = params.column.userProvidedColDef?.cellEditorParams?.maxValue;
+    if (minValue !== undefined) {
+      newValue = Math.max(newValue, minValue);
+    }
+    if (maxValue !== undefined) {
+      newValue = Math.min(newValue, maxValue);
+    }
+    newValue = newValue.toString();
+  }
+  params.data[params.colDef.field] = newValue;
+  return true;
+};
+
+const _enumSetter = (params) => {
+  const enumValues = params.column.userProvidedColDef?.cellEditorParams?.enumValues || [];
+  if (enumValues.length === 0) {
+    console.warn(`Missing enum values for table column "${params.column.colId}"`);
+  }
+
+  let newValue = params.newValue;
+  if (!params.context.editMode || enumValues.indexOf(newValue) === -1) {
+    newValue = params.oldValue;
+  }
+  params.data[params.colDef.field] = newValue;
+  return true;
+};
+
+export const getDefaultColumnsProperties = (onCellChange) => {
+  return {
+    editable: (params) => params.context.editMode,
+    resizable: true,
+    sortable: true,
+    filter: 'agTextColumnFilter',
+    valueSetter: _editModeSetter,
+    onCellValueChanged: (event) => {
+      onCellChange(event);
+    },
+  };
+};
+
+export const getColumnTypes = (dateFormat) => {
+  const _dateComparator = (valueA, valueB, nodeA, nodeB, isInverted) => {
+    return DateUtils.parse(valueA, dateFormat) > DateUtils.parse(valueB, dateFormat) ? 1 : -1;
```

Comment:
Why do you return `-1` and `1`, instead of `true` and `false`?

Reply:
Comparison functions usually return -1, 0 and 1 to describe the 3 possible results of the comparison : less than, equal to or greater than.
In this case, the comparison function is given to AgGrid, and it expects -1, 0 and 1 as returned values.

Reply:
OK, you didn't handle the 0 case, so I was wondering. Thanks.